controlplane/pkg/prefix_pool/prefixpool.go

```diff
@@ -221,24 +250,29 @@ func ExtractPrefix(prefixes []string, prefixLen uint32) (string, []string, error
 			// We found required one.
 			resultPrefix := prefixes[idx]
 			resultPrefixes := append(prefixes[:idx], prefixes[idx+1:]...)
+			/* Verify the allocated addresses does not intersect with one from the excludedPrefix list */
+			err := verifyExcluding(resultPrefix, excludedPrefixes)
+			if err != nil {
+				return "", prefixes, err
+			}
 			// Lets remove from list and return
 			return resultPrefix, resultPrefixes, nil
 		} else {
 			// Update minimal root for split
```

Comment:
Hi, We also need to choose a candidate to split here with verification subnet is fit based on 
verifyExcluding(resultPrefix, excludedPrefixes)